packages/controllers/src/snaps/SnapController.ts

```diff
@@ -387,6 +429,25 @@ type FeatureFlags = {
   dappsCanUpdateSnaps?: true;
 };
 
+type SemVerVersion = string;
+export type CheckSnapBlockListArg = Record<SnapId, SemVerVersion>;
+
+export type CheckSnapBlockListResult = Record<
+  SnapId,
+  {
```

Comment:
A more strict typing that avoids ambiguity could be used, for example
```typescript
export type CheckSnapBlockListResult = Record<SnapId, false | { reason: string, infoUrl?: string }>;
```
That way,  `{ blocked: false, reason: "wat" }` is not possible.